mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.java

```diff
@@ -223,6 +227,10 @@ public void setProtocols(List<Protocol> protocols) {
     this.protocols = protocols;
   }
 
+  public List<Protocol> protocols() {
+    return Collections.unmodifiableList(protocols);
```

Comment:
when we set the protocols we call `Util.immutableList()` so I think unmodifiableList() is not necessary